lib/Store.ts

```diff
@@ -1,35 +1,62 @@
+import { Advertisement, UserData } from "@/types/api";
 import { createClient } from "@supabase/supabase-js";
-import { Database } from "@/types/supabase";
 
 export const SOLD_STATE = 3;
 
 export const database = createClient(
-  process.env.SUPABASE_URL ?? "",
-  process.env.SUPABASE_KEY ?? ""
+  process.env.SUPABASE_URL!,
+  process.env.SUPABASE_KEY!
 );
 
-export type User = Database["public"]["Tables"]["user_data"]["Update"];
-
-export type Advertisement =
-  Database["public"]["Tables"]["advertisement"]["Row"];
-
-export const fetchUserById = async (userId: string) => {
+export const fetchUserById = async (id: string) => {
   try {
     const { data } = await database
       .from("user_data")
-      .select(
-        `user_id, first_name, last_name, class, email,
-        specialization:specialization_id (specialization_name),
-        school:school_id (school_name)`
-      )
-      .eq("user_id", userId);
-
+      .select(`*`)
```

Comment:
Don't we need `specialization_name` and `school_name`?

Reply:
I genuinely don't remember when I edited this function, It was back during summer, try calling the api in your project and see what it returns if it doesn't make sense just leave it as It was before.